### PR TITLE
Feature/remove session context

### PR DESCRIPTION
**Pull request by timsaucer**
This PR removes the `datafusion-python` SessionContext from being owned by `Context` in `datafusion-ray`. This allows the user to use all of the features already existing in `datafusion-python`. 

I have tested it against the example `tips.py`

### CODE REVIEW

File: datafusion_ray/context.py [hunk 1]
```diff
@@ -176,7 +175,7 @@ def execute_query_partition(
     *input_partitions: list[pa.RecordBatch],
 ) -> Iterable[pa.RecordBatch]:
     start_time = time.time()
-    plan = datafusion_ray.deserialize_execution_plan(plan_bytes)
+    # plan = datafusion_ray.deserialize_execution_plan(plan_bytes)
```
Comment (andygrove) [inline datafusion_ray/context.py:178]: nit: can we remove this line now?

File: src/context.rs [hunk 2]
```diff
@@ -44,75 +41,61 @@ type PyResultSet = Vec<PyObject>;
...
-    pub fn plan(&self, sql: &str, py: Python) -> PyResult<PyExecutionGraph> {
-        println!("Planning {}", sql);
-        let df = wait_for_future(py, self.ctx.sql(sql))?;
-        let plan = wait_for_future(py, df.create_physical_plan())?;
+    pub fn plan(&self, plan: &Bound<PyAny>) -> PyResult<PyExecutionGraph> {
+        // println!("Planning {}", sql);
+        // let df = wait_for_future(py, self.ctx.sql(sql))?;
+        // let py_df = self.run_sql(sql, py)?;
+        // let py_plan = py_df.call_method0(py, "execution_plan")?;
+        // let py_plan = py_plan.bind(py);
```
Comment (andygrove) [inline src/context.rs:96]: can we remove this commented out code?

**Review by andygrove — APPROVED** [review]
Amazing. Thank you @timsaucer

**Comment by timsaucer** [comment]
Sorry I missed those commented lines. I’m out for the weekend so I’ll push the edits Monday.

**Comment by andygrove** [comment]
> Sorry I missed those commented lines. I’m out for the weekend so I’ll push the edits Monday.

I think is fine to fix those in a follow up PR

**Pull request merged**